test/x509_time_test.c

```diff
@@ -187,8 +280,53 @@ static int test_x509_cmp_time_current()
     return failed == 0;
 }
 
+static int test_x509_time(int idx)
+{
+    ASN1_TIME *t = NULL;
+    int result;
+
+    if (x509_format_tests[idx].set_string) {
+        /* set-string mode */
+        t = ASN1_TIME_new();
+        if (t == NULL) {
+            TEST_info("test_x509_time(%d) failed: internal error\n", idx);
+            return 0;
+        }
+    }
```

Comment:
Because we need to test if this function can accept a NULL input for the first parameter and if behaves properly. The doc under doc/man3/ASN1_TIME_set.pod didn't describe this usage for the original ASN1_TIME_set_string(), but it's indeed used that way in apps/ca.c and might be used already by other applications. So I think we need to cover this scenario in case of the future-API-internal-change, we would probably get noticed if input of NULL ASN1_TIME  still works or not.